app/app.go

```diff
@@ -1194,22 +1201,23 @@ func (d *Deployments) createDeviceDeploymentWithStatus(
 		return nil, err
 	}
 
-	if err := d.db.InsertDeviceDeployment(ctx, deviceDeployment); err != nil {
+	if err := d.db.InsertDeviceDeployment(ctx, deviceDeployment,
```

Comment:
just for my knowledge - the approach here is to not decrement deployment counter when "deleting" device deployment, but to not increment it, if the device will become a part of the same deployment once again? not sure if this will work in all cases (e.g. after resetting the device, device will not ask for the deployment), but it's probably OK (I think someone who will use this functionality will know what he's doing);
it may be more complex with phased rollouts in the deployments-enterprise, but I'm just guessing here;

Reply:
Correct, we don't decrement the device count in the deployments when we (logically) remove the device deployment records. This is because that counter determines whether the deployment is completed, and this has several side effects. All in all, the data is there, we just mark it as "deleted" so we can ignore it when the device asks for updates, again. This is relevant for dynamic deployments, while it has no practical implications for static deployments. I tested it (manually) bringing these changes to deployments-enterprise and it works correctly.

I want to add that my idea is that we should refactor the retry feature in deployments-enterprise to take advantage of the `deleted` device deployment field, creating a new entry for each try. This will show the different attempts as history independent items in the device deployments history view we are implementing.